meson.build

```diff
@@ -409,107 +412,122 @@ bdb_major_version = ''
 bdb_minor_version = ''
 bdb_version = ''
 
-bdb_dirs = [
-    '/usr/local',
-    '/usr/pkg',
-    '/opt/local',
-    macos_prefix / 'opt/berkeley-db',
-    '/usr',
-]
+if bdb_path != ''
+    bdb_dirs = [ bdb_path ]
+else
+    bdb_dirs = [
+        '/usr/local',
+        '/usr/pkg',
+        '/opt/local',
+        macos_prefix / 'opt/berkeley-db',
+        '/usr',
+    ]
+endif
+
 bdb_subdirs = [
     'db4.6',
-    'db4.7',
-    'db4.8',
-    'db4',
     'db46',
+    'db4.7',
     'db47',
+    'db4.8',
     'db48',
+    'db4',
     'db5.0',
+    'db50',
     'db5.1',
+    'db51',
     'db5.2',
+    'db52',
     'db5.3',
+    'db53',
     'db5',
-    'db50',
-    'db51',
     'db6.1',
+    'db61',
+    'db6.2',
+    'db62',
     '',
 ]
 
-if bdb_path != ''
+if bdb_req_version != ''
+    if not bdb_req_version.version_compare('>=4.6')
+        error('Berkley DB library version is below supported 4.6')
+    endif
+    message('Searching for Berkley DB library version', bdb_req_version)
+endif
+
+foreach dir : bdb_dirs
     foreach subdir : bdb_subdirs
-        if fs.exists(bdb_path / 'include' / subdir / 'db.h')
-            bdb_header += bdb_path / 'include' / subdir / 'db.h'
-            bdb_libdir += bdb_path / 'lib'
-            bdb_includes += include_directories(
-                bdb_path / 'include' / subdir,
-            )
-        endif
-        if bdb_header != ''
-            break
-        endif
-    endforeach
-else
-    foreach dir : bdb_dirs
-        foreach subdir : bdb_subdirs
-            if fs.exists(dir / 'include' / subdir / 'db.h')
-                bdb_header += dir / 'include' / subdir / 'db.h'
-                if target_os == 'sunos' and compiler_64_bit_mode and fs.exists(dir / 'lib/64')
-                    bdb_libdir += dir / 'lib/64'
-                else
-                    bdb_libdir += dir / 'lib'
-                endif
-                bdb_includes += include_directories(
-                    dir / 'include' / subdir,
-                )
+        bdb_include_path = dir / 'include' / subdir
+        bdb_header = bdb_include_path / 'db.h'
+        if fs.exists(bdb_header)
+            bdb_includes = include_directories(bdb_include_path)
+
+            bdb_major_version = run_command(
+                'grep',
+                'DB_VERSION_MAJOR',
+                bdb_header,
+                check: true,
+            ).stdout().strip().substring(25)
+        
+            bdb_minor_version = run_command(
+                'grep',
+                'DB_VERSION_MINOR',
+                bdb_header,
+                check: true,
+            ).stdout().strip().substring(25)
+        
+            bdb_version = bdb_major_version + '.' + bdb_minor_version
+
+            if not bdb_version.version_compare('>=4.6')
+                continue
+            endif
+
+            if bdb_req_version != '' and not bdb_version.version_compare('~' + bdb_req_version)
+                continue
+            endif
```

Comment:
I would like to throw a warning message here, along the lines of "bdb version X requested via -Dwith-bdb-version but not found"

This gives the user more context to the error message that follows.

What do you think?

Reply:
As we search for requested `db` version in cycle over all the dirs and subdirs this would emit a multiple warnings if there are multiple non-matching versions installed. I would rather add this explanation to final error in case particular version was requested but not found.

Reply:
Much better, thanks. This achieves what I wanted to see.